controllers/backplaneconfig_controller.go

```diff
@@ -267,6 +267,13 @@ func (r *MultiClusterEngineReconciler) Reconcile(ctx context.Context, req ctrl.R
 		return result, err
 	}
 
+	if status.AllComponentsReady(r.StatusManager.ReportComponents()) {
```

Comment:
Is it necessary in this case to wait for all components to be ready before removing the hypershift pieces? If there's no dependency it might be better to just clean things up as the component is processed.